src/GraphCoarsen.cpp

```diff
@@ -38,24 +38,16 @@ namespace smoothg
 
   Maybe use the example in parelag-notes.
 */
-GraphCoarsen::GraphCoarsen(const mfem::SparseMatrix& M_proc,
-                           const mfem::SparseMatrix& D_proc,
-                           const GraphTopology& graph_topology)
-    : GraphCoarsen(M_proc, D_proc, nullptr, graph_topology)
-{
-}
-
-GraphCoarsen::GraphCoarsen(const mfem::SparseMatrix& M_proc,
-                           const mfem::SparseMatrix& D_proc,
-                           const mfem::SparseMatrix* W_proc,
-                           const GraphTopology& graph_topology)
+GraphCoarsen::GraphCoarsen(const MixedMatrix& mgL, const GraphTopology& graph_topology)
     :
-    M_proc_(M_proc),
-    D_proc_(D_proc),
-    W_proc_(W_proc),
+    M_proc_(mgL.GetM()),
+    D_proc_(mgL.GetD()),
+    W_proc_(mgL.GetW()),
+    fine_mbuilder_(dynamic_cast<const FineMBuilder*>(&(mgL.GetMBuilder()))),
     graph_topology_(graph_topology),
-    colMapper_(M_proc.Size())
+    colMapper_(M_proc_.Size())
```

Comment:
This should probably be set from `D_proc_.Width()` for consistency.  Is not an issue now but in multilevel `M_proc_` may not be assembled.  Or we can wait till multilevel and fix everything then.

Reply:
I am not sure at this point how should the aggregate submatrix should be obtained in the multilevel setting (which is what `colMapper_` is for), but your suggested change is harmless and simple, so I think it makes sense to do it.